map_model/src/make/transit.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::{HashMap, HashSet};
+
+use anyhow::Result;
+
+use abstutil::Timer;
+use geom::{Distance, Duration, FindClosest, HashablePt2D, Time};
+
+use crate::make::match_points_to_lanes;
+use crate::raw::{RawMap, RawTransitRoute, RawTransitStop};
+use crate::{
+    LaneID, Map, PathConstraints, Position, TransitRoute, TransitRouteID, TransitStop,
+    TransitStopID,
+};
+
+pub fn finalize_transit(map: &mut Map, raw: &RawMap, timer: &mut Timer) {
+    // Snap stops to sidewalks and driving lanes, similar to buildings
+    let mut query: HashSet<HashablePt2D> = HashSet::new();
+    for stop in raw.transit_stops.values() {
+        query.insert(stop.position.to_hashable());
+    }
+    let sidewalk_pts = match_points_to_lanes(
```

Comment:
This is an existing helper method to snap a bunch of points to a sidewalk. Also used for building driveways and parking lots